ADAL/src/ui/ADWebAuthController.m

```diff
@@ -290,6 +290,17 @@ - (BOOL)webAuthShouldStartLoadRequest:(NSURLRequest *)request
         return NO;
     }
     
+    // redirecting to non-https url is not allowed
+    if ([requestURL hasPrefix: @"http"] && ![requestURL hasPrefix: @"https"])
+    {
```

Comment:
just "if ([requestURL hasPrefix: @"http"])" would be sufficient. I could see wanting to specifically block anything that's not either one of the allowed types, redirect URI scheme or HTTPS.